### ISSUE TITLE
ES Decorators downlevel silently breaks on name clash

### ISSUE THREAD

**Issue by robpalme** [microsoft/TypeScript#52324]
# Bug Report

### 🔎 Search Terms

`decorators` `downlevel`

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because Decorators only landed last night in 5.0.0-dev.20230120

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=9&ts=5.0.0-dev.20230120#code/PTAEAEBcEMCcHMCmkBcpEGcBMAGLWAoAgMwFcA7AY0gEsB7c0AfScwBFFK7ZpJEAKGAmRpo5AJ4AaZl3J8AHqlBjxASlETQAbwKg9oWMlKxGZKrQahBACxoYNU0ADoXcePeUTV23fr+yMOgAbRCcgunh+ACIOLh4+ABMo71AQUAAhAFUAcTRyRAA3RFhQSmggkITQFABafl8-PUNIY0YhJEgnMoqbO2kXJzcMVQBuBtAAXzGJokog6AwMUABZcQBhecWfP3AWdk5uXkRxjBhaSlAAW2RrOgT+by0ZmYIgA)

### 💻 Code

```ts
// @target: es2022

function __esDecorate(target: any, _context: any) {
    return function (this: any, ...args: any) {
        console.log("Decorated")  // BUG: never called :-(
        return target.call(this, ...args);
    };
}

class MyClass {
    @__esDecorate
    static method() {}
}
```

### 🙁 Actual behavior

The decorator is never called because the identifier `__esDecorate` clashes with a magic name used by the down-level transpilation.

Note that this works correctly if you disable down-levelling via `target: ESNext`

### 🙂 Expected behavior

Only when down-levelling is enabled, the checker should error and say you are not allowed to use that identifier.

**Comment by DanielRosenwasser** [comment]
@robpalme did you actually run into this in your codebase?

**Comment by rbuckton** [comment]
Yes, this is intentional and has been a part of TS since the beginning. It's fairly rare to encounter a conflict. We could potentially emit an error if the helper doesn't match the signature we intend to use. We do something similar to verify the private field helpers when we use the ones from `tslib`.

**Comment by robpalme** [comment]
>@robpalme did you actually run into this in your codebase?

Nope - this was just me trying to break it 😉

If this is intentional, so be it. The oddity is that this gives a clear semantic difference depending on whether downlevelling is enabled.

**Comment by rbuckton** [comment]
> If this is intentional, so be it. The oddity is that this gives a clear semantic difference depending on whether downlevelling is enabled.

That's always been the case. I believe @bterlson used to use that capability to replace the default `__async` helper in `ecmarkup` with one that used RSVP, though that's since been removed as `ecmarkup` eventually switched to native async function emit.
